test/utils/Addresses.sol

```diff
@@ -1,23 +1,45 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.7.6;
 
-import "forge-std/Test.sol";
+import { Test, Vm } from "forge-std/Test.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { MockERC20 } from "test/mocks/MockERC20.sol";
+import { MockWETH } from "test/mocks/MockWETH.sol";
 
 address constant ETH_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
 address constant ZERO_ADDRESS = address(0);
 
 contract Addresses is Test {
+    IERC20 public weth;
+    IERC20 public usdt;
+    IERC20 public usdc;
+    IERC20 public dai;
+    IERC20 public wbtc;
+    IERC20 public lon;
+    IERC20 public ankreth;
+    IERC20[] public tokens;
+
     string private file = readAddresses(vm);
 
-    address WETH_ADDRESS = abi.decode(vm.parseJson(file, "$.WETH_ADDRESS"), (address));
-    address USDT_ADDRESS = abi.decode(vm.parseJson(file, "$.USDT_ADDRESS"), (address));
-    address USDC_ADDRESS = abi.decode(vm.parseJson(file, "$.USDC_ADDRESS"), (address));
+    // Since token may be newly deployed in local testnet case, these imported addresses may not be the actual ones.
+    address private WETH_ADDRESS = abi.decode(vm.parseJson(file, "$.WETH_ADDRESS"), (address));
```

Comment:
I know that it mixes `private` and `public` among these addresses but currently I don't have a better solution.

Reply:
Do you mean that `TOKEN_ADDRESS` is private, and `token` is public? What goal would you want to achieve?

Reply:
Do you mean that `TOKEN_ADDRESS` is private, and `token` is public? What goal would you want to achieve?

Reply:
Yeah, the goal of doing this is to unify the way of reading token address. For example a test can read WETH address by `WETH_ADDRESS` or `address(weth). But for local test, tokens are new deployed mock tokens so `WETH_ADDRESS` is not the read address of mock WETH.

Reply:
However, this only applies to some of tokens so there're private and public `TOKEN_ADDRESS`.